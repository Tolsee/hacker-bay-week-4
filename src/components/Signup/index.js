// @flow

import React from 'react';
import styled from 'styled-components';

import {
  Form,
  Button,
  Input,
  Icon
} from 'antd';

const { Item: FormItem } = Form;

const Heading = styled.h2`
  width: 100%;
  text-align: center;
`;

const LoginButton = styled(Button)`
  width: 100%;
`;

export default class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      userName: '',
    };
  }

  emitEmpty = () => {
    this.userNameInput.focus();
    this.setState({ userName: '' });
  }

  onChangeUserName = (e) => {
    this.setState({ userName: e.target.value });
  }

  render() {
    const { userName } = this.state;
    const suffix = userName ? <Icon type="close-circle" onClick={this.emitEmpty} /> : null;
    return (
      <React.Fragment>
        <Heading>Signup</Heading>
        <FormItem>
          <Input
            placeholder="Enter your email"
            prefix={<Icon type="mail" style={{ color: 'rgba(0,0,0,.25)' }} />}
            suffix={suffix}
            value={userName}
            onChange={this.onChangeUserName}
            ref={node => this.userNameInput = node}
          />
        </FormItem>
        <FormItem>
          <Input
            placeholder="Enter your password"
            prefix={<Icon type="lock" style={{ color: 'rgba(0,0,0,.25)' }} />}
            suffix={suffix}
            value={userName}
            onChange={this.onChangeUserName}
            ref={node => this.userNameInput = node}
          />
        </FormItem>
        <FormItem>
          <LoginButton type="primary" htmlType="submit" className="login-form-button">
           Singup
          </LoginButton>
          Already have an account <a href="/login">Login</a>
        </FormItem>
      </React.Fragment>
    );
  }
}

