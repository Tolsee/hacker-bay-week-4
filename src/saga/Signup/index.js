import { takeLatest } from 'redux-saga';
import { call } from 'redux-saga/effects';

import { Request } from 'saga/index';
import { SIGNUP, signupSuccess, signupError } from 'actions';

function* signup(action) {
  console.log('Action watched', action);
  yield call(Request.post('/user/signup', signupSuccess, signupError, action.payload));
}

export default function* signupSage() {
  yield takeLatest(SIGNUP, signup);
};