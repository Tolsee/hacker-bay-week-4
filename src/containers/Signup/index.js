import React, { Component } from 'react';
import { connect } from 'react-redux';

import Signup from 'components/Signup';

import { signup } from 'actions';

const mapStateToProps = state => ({

});

const mapDispatchToProps = dispatch => ({
  signup: (data) => dispatch(signup(data))
});

type Props = {
  signup: Function
};

class SignupContainer extends Component<Props> {
  signup = values => {
    console.log(values);
    this.props.signup(values);
  };

  render() {
    console.log('Props: ', this.props);
    return (
      <Signup {...this.props} onSubmit={this.signup}/>
    )
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SignupContainer);