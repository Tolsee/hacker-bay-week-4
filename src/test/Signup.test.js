import React from 'react';
import { shallow } from 'enzyme';
import Signup from 'components/Signup';

it('should render login form correctly', () => {
  const component = shallow(<Signup />);

  expect(component).toMatchSnapshot();
});