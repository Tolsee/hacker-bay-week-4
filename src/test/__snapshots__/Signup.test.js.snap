// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login form correctly 1`] = `
<Connect(Form(Signup))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="signup-form"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
